src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/processors/date/view_model.js

```diff
@@ -1,14 +1,19 @@
+import _ from 'lodash';
 import Processor from '../base/view_model';
 
 export class Date extends Processor {
-  constructor(processorId) {
+  constructor(processorId, oldProcessor) {
     super(processorId, 'date', 'Date');
-    this.sourceField = '';
-    this.targetField = '@timestamp';
-    this.formats = [];
-    this.timezone = 'Etc/UTC';
-    this.locale = 'ENGLISH';
-    this.customFormat = '';
+    _.assign(this,
```

Comment:
Nit, but it might make intentions more clear if you use _.defaults here instead of _.assign since you're basically setting defaults.